hugegraph-server/hugegraph-dist/src/assembly/travis/test-check-port.sh

```
@@ -0,0 +1,407 @@
+    && pass "/dev/tcp+timeout: connection refused (exit 1) → port free → exit 0" \
+    || fail "/dev/tcp+timeout: connection refused → expected exit 0, got 1"
+
+section "/dev/tcp fallback — real loopback (no mock)"
+
+(
+    if ! command -v python3 >/dev/null 2>&1; then
+        echo "SKIP: python3 not available — skipping real loopback test"
+        exit 0
+    fi
+
+    source "$UTIL_SH"
```

Shellcheck (0.11.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)